Shell/FOOLElimination.cpp

```diff
@@ -374,9 +393,9 @@ void FOOLElimination::process(TermList ts, Context context, TermList& termResult
   if (context == FORMULA_CONTEXT) {
     return;
   }
-
   // preprocessing of the term does not affect the sort
-  ASS_EQ(sort, SortHelper::getResultSort(termResult, _varSorts));
+  //ASS_EQ(sort, SortHelper::getResultSort(termResult, _varSorts));
+  //TODO assert that it is a variant
```

Comment:
Can we do it easily, these days?

Reply:
It doesn't seem so. The closest we have is `MatchingUtils::isVariant`, but this function works on literals, not on TermLists. It seems a bit overkill to re-engineer the function for the sake of a single assertion.

Reply:
OK!